sonic_platform_base/sonic_xcvr/api/public/cmisCDB.py

```diff
@@ -0,0 +1,430 @@
+            20h-2Fh=Reserved
+            30h-3Fh=Custom
+
+        FAILED
+            00h=Reserved
+            01h=CMDCode unknown
+            02h=Parameter range error or parameter not supported
+            03h=Previous CMD was not ABORTED by CMD Abort
+            04h=Command checking time out
+            05h=CdbCheckCode Error
+            06h=Password related error (command specific meaning)
+            07h=Command not compatible with operating status
+            08h-0Fh=Reserved for STS command checking error
+            10h-1Fh=Reserved
+            20h-2Fh=For individual STS command or task error
+            30h-3Fh=Custom
+        '''
+        status = self.xcvr_eeprom.read(consts.CDB1_STATUS)
+        is_busy =  bool((status >> 7) & 0x1)
+        cnt = 0
+        while is_busy and cnt < MAX_WAIT:
+            time.sleep(1)
+            is_busy =  bool((status >> 7) & 0x1)
+            cnt += 1
+        return status
+
+    def write_cdb(self, cmd):
+        '''
+        This function writes a CDB command to page 0x9f
+        '''
+        self.xcvr_eeprom.write_raw(LPLPAGE*PAGE_LENGTH+CDB_WRITE_MSG_START, len(cmd)-CMDLEN, cmd[CMDLEN:])
+        self.xcvr_eeprom.write_raw(LPLPAGE*PAGE_LENGTH+INIT_OFFSET, CMDLEN, cmd[:CMDLEN])
+
+    def read_cdb(self):
+        '''
+        This function reads the reply of a CDB command from page 0x9f
+        '''
+        rpllen = self.xcvr_eeprom.read(consts.CDB_RPL_LENGTH)
+        rpl_chkcode = self.xcvr_eeprom.read(consts.CDB_RPL_CHKCODE)
+        rpl = self.xcvr_eeprom.read_raw(LPLPAGE*PAGE_LENGTH+CDB_RPL_OFFSET, rpllen)
+        return rpllen, rpl_chkcode, rpl
+
+    # Query status
+    def cmd0000h(self):
+        '''
+        This QUERY Status command may be used to retrieve the password acceptance
+        status and to perform a test of the CDB interface
+        '''
+        cmd = bytearray(b'\x00\x00\x00\x00\x02\x00\x00\x00\x00\x10')
+        cmd[133-INIT_OFFSET] = self.cdb_chkcode(cmd)
+        for attemp in range(MAX_TRY):
+            self.write_cdb(cmd)
+            time.sleep(2)
+            status = self.cdb1_chkstatus()
+            if (status != 0x1):
+                logger.info('CDB1 status: Fail. CDB1 status %d' %status)
+                continue
+            else:
+                break
+        return self.read_cdb()
+
+    # Enter password
+    def module_enter_password(self):
+        '''
+        The Enter Password command allows the host to enter a host password
+        '''
+        cmd = bytearray(b'\x00\x01\x00\x00\x04\x00\x00\x00\x00\x00\x10\x11')
+        cmd[133-INIT_OFFSET] = self.cdb_chkcode(cmd)
+        for attemp in range(MAX_TRY):
+            self.write_cdb(cmd)
+            time.sleep(2)
+            status = self.cdb1_chkstatus()
+            if (status != 0x1):
+                logger.info('CDB1 status: Fail. CDB1 status %d' %status)
+                continue
+            else:
+                break
+        return status
+
+    def cmd0040h(self):
```

Comment:
please rename to meaningful function name

Reply:
done